alacritty_terminal/src/term/mod.rs

```diff
@@ -1170,13 +1097,21 @@ impl<T> Term<T> {
     #[inline]
     pub fn toggle_vi_mode(&mut self) {
         self.mode ^= TermMode::VI;
-        self.selection = None;
 
-        // Reset vi mode cursor position to match primary cursor.
-        if self.mode.contains(TermMode::VI) {
+        let vi_mode = self.mode.contains(TermMode::VI);
+
+        // Do not clear selection when entering search.
+        if self.regex_search.is_none() || !vi_mode {
```

Comment:
Wouldn't it be cleared anyway because you 'type'? I don't quite get it.
Or what is the idea behind this code, I don't feel like I get it.

Reply:
Typing into the search bar does not clear the selection, no.

So you can start a selection, search, then your selection will be updated to the new point.

Reply:
I mean that you can't select while search bar is active, and entering search clears selection anyway. If you leave search you won't have a selection anyway. If you can give me an example how to enter search and have selection, it'll be really helpful to understand the intent of that.

Reply:
> entering search clears selection anyway.

It shouldn't. Why would it?